src/transformers/integrations/peft.py

```diff
@@ -517,3 +517,59 @@ def _dispatch_accelerate_model(
             offload_dir=offload_folder,
             **dispatch_model_kwargs,
         )
+
+    def delete_adapters(self, adapter_names: Union[List[str], str]):
+        """
+        Delete an adapter's LoRA layers from the underlying model.
+    
+        Args:
+            adapter_names (`Union[List[str], str]`):
+                The name(s) of the adapter(s) to delete.
+    
+        Example:
+    
+        ```py
+        from diffusers import AutoPipelineForText2Image
+        import torch
+    
+        pipeline = AutoPipelineForText2Image.from_pretrained(
+            "stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16
+        ).to("cuda")
+        pipeline.load_lora_weights(
+            "jbilcke-hf/sdxl-cinematic-1", weight_name="pytorch_lora_weights.safetensors", adapter_names="cinematic"
+        )
+        pipeline.delete_adapters("cinematic")
+        ```
+        """
+
+        check_peft_version(min_version=MIN_PEFT_VERSION)
+
+        if not is_peft_available():
+            raise ImportError("PEFT is not available. Please install PEFT to use this function: `pip install peft`.")
+
+        if not self._hf_peft_config_loaded:
+            raise ValueError("No adapter loaded. Please load an adapter first.")
+
+        from peft.tuners.tuners_utils import BaseTunerLayer
+    
+        if isinstance(adapter_names, str):
+            adapter_names = [adapter_names]
+    
```

Comment:
Before starting to delete the adapters, how about checking that each adapter name is present in the `self.peft_config` and raise an error if it's not present?

Reply:
Done